client/src/hooks/useFindTrip.tsx

```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from 'react';
+import { TripEntry, getTrip, placeholder } from '../lib/data';
+
+export default function useFindTrip(userId: number, tripId: number): TripEntry {
+  const [trip, setTrip] = useState<TripEntry>(placeholder);
+
+  useEffect(() => {
+    async function readTrips(): Promise<void> {
+      try {
+        const activeTrip = await getTrip(userId, tripId);
+        if (activeTrip.length > 0) setTrip(activeTrip[0]);
+      } catch (error) {
+        console.error('Error getting trips', error);
```

Comment:
By just catching and logging the error, you are making it impossible to show the user that there was an error.

The best way to do error handling in a hook like this is to return a state object from the hook instead of just the TripEntry.

I usually do something like this:
```
type FindTripState = {
  trips?: TripEntry[];
  isLoading: boolean;
  error: Error;
};
export function useFindTrip(userId: number, tripId: number): FindTripState {
  // useState for trips and useState for error
  // useEffect that sets either trips or error depending on what happens
  return { trips, error, isLoading: !trips && !error };
}
```
`isLoading` is true if there is not a trip and not an error, since one or the other will always get set in the useEffect.

Then in the caller, you destructure all of these and can display either a Loading message or an Error message if necessary, and the `trips` if all went well. (TypeScript will think `trips` can be undefined even if isLoading is false because it doesn't look at this code. So I usually handle this with:
```
if (isLoading) { /* return loading message */ }
if (error || !trips) { /* return error message */ }
```
This way TS knows that in any code after those checks that `trips` is defined.